Bugfix/#1256 cannot navigate within project folder

NickyWeber opened the pull request:
Fixes https://github.com/spritebuilder/SpriteBuilder/issues/1256

NickyWeber committed:
Unregistered .spritebuilder extension.

NickyWeber committed:
Move Spritebuilder.entitlements to Support Files group.
Removed Spritebuilder.entitlements duplicate reference.

NickyWeber committed:
File extension .ccbproj registered again for Spritebuilder.
If a .ccbproj file opened Spritebuilder will pop up an open dialogue to ask for user's consent to access the project folder.

NickyWeber committed:
Merge branch 'develop' into bugfix/#1256_cannot_navigate_within_project_folder

cocojoe commented:
So I cleaned out my existing SpriteBuilder extension handlers first and installed a release build into applications.  If anyone is interested you can find all the registered instances with,  often many exist that are no longer even installed.

`/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -dump | grep spritebuilder -i | grep path | less`

cocojoe commented:
Navigation into `project.spritebuilder` all good, I can double click `project.ccbproj` which opens up SB and presents you with a sandbox dialog to grant permission and open the `project.spritebuilder`

![screenshot 2015-01-07 12 54 21](https://cloud.githubusercontent.com/assets/928115/5645830/511758b0-966c-11e4-9b3a-f1d9fe98c7cf.png)

This behaviour seems fine to me, the default user action is to open in case anyone blindly hits enter :)

Only comment would be the `project.ccbproj` icon is now a terminal exec icon.

![screenshot 2015-01-07 12 56 30](https://cloud.githubusercontent.com/assets/928115/5645847/9f8e760e-966c-11e4-89c2-bcb5c696d74e.png)

NickyWeber committed:
Security scoped bookmark added in case a user opens the project with the project file.
The bookmark store the access to the project folder. As soon as the open dialog to grant access to the project folder is dismissed successfully the bookmark is stored in the user defaults.

NickyWeber commented:
Added security scoped bookmarks.

Terminal icon too here, running yosemite. It's weired: It's all configured correctly in XCode.

`/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -dump`

Contains the correct entries, the icns file is copied to the Resources folder of the app.

I killed and restarted the Launch Services, restatert my computer as well:
`/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister -kill -r -domain local -domain system -domain user`
 The wrong icon is persisting.

In case someone needs to remove mutliple entries of the Spriterbuilder.app from the launch services db:

`/System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -dump | grep "/SpriteBuilder.app" | sed 's/     path:          //g' | xargs /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/Support/lsregister -u`

![bildschirmfoto 2015-01-07 um 14 39 46](https://cloud.githubusercontent.com/assets/5259653/5646330/41efe65e-967b-11e4-9b54-4c2976c54f50.png)

NickyWeber committed:
Error log message was incomplete.
